yesod-form/Yesod/Form/Fields.hs

```diff
@@ -768,6 +835,22 @@ selectFieldHelper outside onOpt inside opts' = Field
             x -> case olReadExternal opts x of
                     Nothing -> Left $ SomeMessage $ MsgInvalidEntry x
                     Just y -> Right $ Just y
+-- |
+-- A helper function for constucting 'selectField's. You may want to use this when you define your custom 'selectField's or 'radioField's.
+selectFieldHelperGrouped
+        :: (Eq a, RenderMessage site FormMessage)
+        => (Text -> Text -> [(Text, Text)] -> WidgetFor site () -> WidgetFor site ()) -- ^ Outermost part of the field
+        -> (Text -> Text -> Bool -> WidgetFor site ()) -- ^ An option for None if the field is optional
+        -> (Text -> Text -> [(Text, Text)] -> Text -> Bool -> Text -> WidgetFor site ()) -- ^ Other options
+        -> (Text -> WidgetFor site ()) -- ^ Group headers placed inbetween options
+        -> HandlerFor site (OptionList a)
+        -> Field (HandlerFor site) a
+selectFieldHelperGrouped outside onOpt inside group opts' =
```

Comment:
@snoyberg can you give me a hint here?

When I pass `optsFlat` instead of `opts'` as last parameter (both have type `OptionList`), I get this error:

```
/home/jakob/projects/yesod/yesod-form/Yesod/Form/Fields.hs:849:44: error:
    • Couldn't match expected type ‘HandlerFor site (OptionList a)’
                  with actual type ‘Field m0 (OptionList a)’
    • In the fourth argument of ‘selectFieldHelper’, namely
        ‘optsFlat
```
I don't know how to fmap the `OptionList` while "keeping the surrounding monad"...